### PR TITLE
Fixes #30288 - replace CPU/Cores/socket input to react comp

### PR DESCRIPTION
**Pull request by yifatmakias**

### CODE REVIEW

File: webpack/assets/javascripts/react_app/components/CPUCoresInput/cpuCoresInput.scss [hunk 1]
```diff
@@ -1,6 +1,12 @@
...
-.help-block-icon {
+.warning-icon {
   margin-right: 3px;
+  color: #f0ab00;
+}
+
+.error-icon {
+  margin-right: 3px;
+  color: #c9190b;
 }
```
Comment (MariaAga) [inline webpack/assets/javascripts/react_app/components/CPUCoresInput/cpuCoresInput.scss]: Can you use the colors from this file?
https://github.com/theforeman/foreman/blob/develop/webpack/assets/javascripts/react_app/common/colors.scss

File: webpack/assets/javascripts/react_app/components/common/forms/NumericInput.js [hunk 2]
```diff
@@ -21,25 +25,37 @@ const NumericInput = ({
       value={value}
       precision={precision}
       onChange={onChange}
+      disabled={disabled}
+      name={name}
+      id={id}
+      ClassName={inputClassName}
```
Comment (MariaAga) [inline webpack/assets/javascripts/react_app/components/common/forms/NumericInput.js]: should be `className`

File: webpack/assets/javascripts/react_app/components/CPUCoresInput/CPUCoresInput.js [hunk 3]
```diff
@@ -36,38 +40,51 @@ const CPUCoresInput = ({
...
+            disabled={disabled}
+            name={name}
+            id={id}
+            inputClassName={inputClassName}
+          />
+        </Col>
+      </Row>
+      <Row>
+        <Col className="col-md-2" />
+        <Col className="col-md-4">
```
Comment (MariaAga) [inline webpack/assets/javascripts/react_app/components/CPUCoresInput/CPUCoresInput.js]: Should be `<Col md={2}>` (or 4)

File: webpack/assets/javascripts/react_app/components/CPUCoresInput/CPUCoresInput.js [hunk 4]
```diff
@@ -36,38 +40,51 @@ const CPUCoresInput = ({
...
-        )}
-      </Grid>
-    </FormGroup>
+    <Grid>
+      <Row>
+        <Col>
+          <NumericInput
+            value={value}
+            minValue={minValue}
+            onChange={v => handleChange(v)}
```
Comment (MariaAga) [inline webpack/assets/javascripts/react_app/components/CPUCoresInput/CPUCoresInput.js]: Can be just  `onChange={handleChange}`

File: webpack/assets/javascripts/react_app/components/CPUCoresInput/CPUCoresInput.js (hunk 3 above)
Comment (MariaAga) [inline webpack/assets/javascripts/react_app/components/CPUCoresInput/CPUCoresInput.js]: Also I think for the warnings it makes sense to take the rest of the Row so can you change 4 to 10?

**Review by MariaAga — CHANGES_REQUESTED** [review]
Thanks Yifat! Snapshots should be replaced and can you add tests for the warning and error states of the component?
Also I think there shouldn't be space between the warning and the cpu input, and instead there should be space between the warning and the next input, you can maybe add a class to the cpu input and add/remove margins

**Comment by yifatmakias** [comment]
@MariaAga Thanks for the review.
I fixed the tests and all the comments.
I still need to add more tests and to fix the double space between the input to the alert message.

**Comment by amirfefer** [comment]
foreman tests failures look related

**Comment by yifatmakias** [comment]
@amirfefer Yes I know, I am working on fixing it.

File: webpack/assets/javascripts/react_app/components/CPUCoresInput/__tests__/CPUCoresInput.test.js [hunk 5]
```diff
@@ -5,13 +5,39 @@ const props = {
...
+  'should render with warning message': {
+    ...warningProps,
+  },
+};
+
+const errorFixtures = {
+  'should render with warning message': {
+    ...errorProps,
+  },
+};
```
Comment (MariaAga) [inline webpack/assets/javascripts/react_app/components/CPUCoresInput/__tests__/CPUCoresInput.test.js]: Can you merge this into 1 object? 
`fixtures ={ 'should..default...':{},  'should..warning...':{}, 'should..error...':{}}`

**Review by MariaAga — DISMISSED** [review]
Looks like the tests are working now :tada:

**Comment by yifatmakias** [comment]
[test foreman]

**Comment by yifatmakias** [comment]
[test katello]

**Comment by yifatmakias** [comment]
This PR depends on other PR:
https://github.com/theforeman/foreman/pull/7896

**Comment by yifatmakias** [comment]
[test katello]

**Comment by yifatmakias** [comment]
[test katello]

**Comment by yifatmakias** [comment]
[test katello]

**Comment by yifatmakias** [comment]
[test katello]

**Comment by yifatmakias** [comment]
[test katello]

**Comment by yifatmakias** [comment]
[test katello]

**Comment by yifatmakias** [comment]
[test katello]

**Comment by yifatmakias** [comment]
[test foreman]

**Review by xprazak2 — DISMISSED** [review]
Works well on libvirt, I haven't tested ovirt or vmware.

**Review by shiramax — CHANGES_REQUESTED** [review]
Thanks @yifatmakias 

I see some strange behavior here, every time I increase the socket, the cores also increase and vice versa, 
it shouldn't be like that, the cores and socket are completely different attributes.

**Comment by yifatmakias** [comment]
@shiramax You are definitely right I will figure out the problem and fix this.
Thanks!

**Comment by yifatmakias** [comment]
[test katello]

**Review by ezr-ondrej — APPROVED** [review]
This works and code looks good.
Thanks a lot, loving this :heart: :+1:

**Comment by ezr-ondrej** [comment]
@MariaAga, @shiramax were your concerns addressed?

**Review by shiramax — APPROVED** [review]

**Comment by shiramax** [comment]
@ezr-ondrej yes. thanks

**Pull request merged**